internal/ui/backup.go

```diff
@@ -92,6 +93,9 @@ func (b *Backup) Run(ctx context.Context) error {
 	for {
 		select {
 		case <-ctx.Done():
+			if errors > 0 {
```

Comment:
This error check is, in my opinion, in the wrong place. The status UI should not decide whether an error in the archiver code has occurred or not. Or using a different terminology: The business logic should be separated from the user interface code. The need to modify both the normal text and the JSON output code is also a symptom of this problem. The fact that the error is effectively reported via a side-channel makes the control flow rather hard to follow.

I'd propose that the error check should happen here, by wrapping the `Error` function:

https://github.com/restic/restic/blob/c03bc88b297f960868d171e098771255f6f11be1/cmd/restic/cmd_backup.go#L560